wildmatch.go

```diff
@@ -73,14 +74,73 @@ func NewWildmatch(p string, opts ...opt) *Wildmatch {
 		w.p = strings.ToLower(w.p)
 	}
 
-	w.ts = parseTokens(
-		strings.Split(w.p, string(filepath.Separator)),
-		w.matchPathname,
-	)
+	w.ts = parseTokens(split(w.p), w.matchPathname)
 
 	return w
 }
 
+// split splits a filepath "p" into its component parts (based on sep). It
+// behaves similarly to the following:
+//
+//   strings.Split(p, string(sep))
+//
+// but retains `\` (including on Windows) when followed by an escapable
+// character.
+//
+// For example, the following splits are valid:
+//
+//   "foo/bar/baz"   -> ["foo", "bar", "baz"]
+//   "foo/bar\*/baz" -> ["foo", "bar\*", "baz"]
+//   "foo\bar\*\baz" -> ["foo", "bar\*", "baz"]
+//   ""              -> []
+//   "/"             -> [""]
+func split(p string) []string {
```

Comment:
Would it make sense to transform these examples in a test case?

Reply:
@larsxschneider Good idea! I started doing this, and then realized that the existence of `split` is unnecessary, since it is formally equivalent to `strings.Split(x, string(sep))`, since we assume that Windows shells automatically translate to POSIX-style paths. See: https://github.com/git-lfs/wildmatch/commit/fd9202b98e893cc00e05c2552dec3c15e3bf3841.